cli_tools/common/utils/logging/firelog_logger.go

```diff
@@ -0,0 +1,311 @@
+//  Copyright 2019 Google Inc. All Rights Reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package logging
+
+import (
+	"bytes"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	daisyutils "github.com/GoogleCloudPlatform/compute-image-tools/cli_tools/common/utils/daisy"
+	"github.com/GoogleCloudPlatform/compute-image-tools/daisy"
+	"github.com/google/uuid"
+	"github.com/minio/highwayhash"
+)
+
+var (
+	httpClient            httpClientInterface = &http.Client{Timeout: 5 * time.Second}
+	serverURL                                 = serverURLProd
+	serverLogEnabled                          = false
+	logMutex                                  = sync.Mutex{}
+	nextRequestWaitMillis int64
+)
+
+// constants used by logging
+const (
+	serverURLProd        = "https://firebaselogging-pa.googleapis.com/v1/firelog/legacy/log"
```

Comment:
Please consider obfuscating using a string interleave--not to confuse humans but to make it more difficult for automation to discover the URL and throw requests at it.

Example: https://github.com/firebase/firebase-ios-sdk/blob/7ac9dbcc9e9ddec139051612c71ec01059ddbff1/GoogleDataTransportCCTSupport/GDTCCTLibrary/GDTCCTUploader.m#L69